File: ftc_app-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/OpModes/Competition/BlueAuto.java
package org.firstinspires.ftc.teamcode.OpModes.Competition;

import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;

import org.firstinspires.ftc.robotcore.external.navigation.RelicRecoveryVuMark;
import org.firstinspires.ftc.teamcode.Robots.MecanumRobot;

@Autonomous(name = "Blue Autonomous")
public class BlueAuto extends LinearOpMode {

    @Override public void runOpMode() throws InterruptedException {
        MecanumRobot robot = new MecanumRobot();
        robot.setup(this);

        waitForStart();

        // Detect the crypto box to put the glyph in.
        while(robot.identifier.identifyPictograph() == RelicRecoveryVuMark.UNKNOWN && opModeIsActive()) {
            idle();
        }

        RelicRecoveryVuMark pictograph = robot.identifier.identifyPictograph();

        // Detect the color of the jewels

        // Knock off the jewel

        // Drive to the crypto boxes

        // Detect the column, drive to it and deliver glyph
        switch(pictograph) {
            case LEFT: break;
            case CENTER: break;
            case RIGHT: break;

            default: break; // This should never happen. If it does we need to display an error.
        }
    }

}

File: ftc_app-3.5/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/competition/BlueAuto.kt
package org.firstinspires.ftc.teamcode.opmodes.competition

import com.qualcomm.robotcore.eventloop.opmode.Autonomous
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode
import org.firstinspires.ftc.teamcode.robots.RelicRecoveryRobot

@Autonomous(name = "Blue Autonomous")
class BlueAuto : LinearOpMode() {

    @Throws(InterruptedException::class)
    override fun runOpMode() {
        val robot = RelicRecoveryRobot()
        robot.setup(this)

        waitForStart()

        robot.lowerJewelStick()
        sleep(1500)

        val colorSensor = robot.colorSensor
        while(opModeIsActive()) {
            if (colorSensor.red() > colorSensor.blue()) {
                robot.turn(0.5, -15)
                sleep(1000)
                robot.raiseJewelStick()
                robot.turn(0.5, 15)
                break
            } else if (colorSensor.red() < colorSensor.blue()) {
                robot.turn(0.5, 15)
                sleep(1000)
                robot.raiseJewelStick()
                robot.turn(0.5, -15)
                break
            }
        }

        val power = Math.sqrt(2.0)/2
        robot.setDirection(power, power, 0.0)
        sleep(2500)

        robot.stop()
    }

}
